src/messageComposer/middleware/pollComposer/state.ts

```diff
@@ -116,24 +116,23 @@ export const pollCompositionStateProcessors: Partial<
     const { index, text } = value;
     const prevOptions = data.options || [];
 
-    const shouldAddEmptyOption =
-      prevOptions.length < MAX_POLL_OPTIONS &&
-      (!prevOptions || (prevOptions.slice(index + 1).length === 0 && !!text));
-
     const shouldRemoveOption =
       prevOptions && prevOptions.slice(index + 1).length > 0 && !text;
 
     const optionListHead = prevOptions.slice(0, index);
-    const optionListTail = shouldAddEmptyOption
-      ? [{ id: generateUUIDv4(), text: '' }]
-      : prevOptions.slice(index + 1);
+    const optionListTail = prevOptions.slice(index + 1);
 
     const newOptions = [
       ...optionListHead,
       ...(shouldRemoveOption ? [] : [{ ...prevOptions[index], text }]),
       ...optionListTail,
     ];
 
+    // Add new option if all the options are filled
+    if (!newOptions.some((option) => !option.text.trim())) {
```

Comment:
I would prefer to have a variable that says what the condition is checking. Like for example `shouldAddEmptyOption`. It helps understanding the code.

Reply:
Also, we should be checking whether we are not reaching `MAX_POLL_OPTIONS` limit - hard limit from BE (`prevOptions.length < MAX_POLL_OPTIONS`)

Reply:
I have pushed the suggestions. Please let me know what do you think. Thanks 😄